### PR TITLE
(fix/RestructuredComponents)-Restructured and Routed Components

### LINKED ISSUE #46: Change Structure of components
Change the structure of components, add all the current components and call it homepage, 
Make 3 new folders, 
-> codemap_details
-> oppc_details
-> one_codemap 

Make sure all the imports are correct after making the changes.

### PR DESCRIPTION
**Pull request by harsha14gupta**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

Fixed #46 
Added dependencies:
react-router-dom": "5.2.0"
"react-scroll": "1.8.0"

## Type of change

Please check options that are relevant to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have squashed my commits
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by Manvityagi — APPROVED** [review]

File: src/homepage/Components/Impact.js [hunk 1]
```diff
@@ -14,45 +14,47 @@ export default class Impact extends Component {
...
+          link: "/codemaps"
         },
         {
           id: 3,
           name: "Open-Source Journey",
           imgUrl: "image3.svg",
           para1:
             "Get your journey in open-source started with our open-source projects!",
-          para2: "",
+          para2: ""
```
Comment (vinitshahdeo) [inline src/homepage/Components/Impact.js]: @harsha14gupta Though it's an optional change but ideally we should keep such hardcoded JSON out of the component. Can we have `ImpactData.js` file outside and import it here?

File: src/homepage/Components/Banner.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 import { Container, Row, Col, Jumbotron, Image } from "react-bootstrap";
 import code from "../assets/code.svg";
 import girl from "../assets/girl.svg";
 import smallscreen from "../assets/smallscreen.png";
 
 export default function Banner() {
   return (
-    <Jumbotron className={styles.Jumbotron} fluid>
+    <Jumbotron id="home" className={styles.Jumbotron} fluid>
       <br />
```
Comment (vinitshahdeo) [inline src/homepage/Components/Banner.js]: @harsha14gupta Why there're many `<br />` statements? Can't we have CSS for `padding-bottom` / `margin-bottom`?

**Review by vinitshahdeo — CHANGES_REQUESTED** [review]
@harsha14gupta Added a few more comments.

**Review by vinitshahdeo — APPROVED** [review]
Good job @harsha14gupta 👏

**Pull request merged**